Fujitsu general climate compilation error

<!-- Thanks for reporting a bug for this project. READ THIS FIRST:

- Provide as many details as possible. Simply saying "X gives bug" or "X gives error" is not enough!
- Paste logs, configuration sample and code into the backticks (```).
- Read through the template carefully and fill out all missing details.
- Please also search for similar issues in this issue tracker first and read through the ESPHome FAQ.

DO NOT DELETE ANY TEXT from this template! Otherwise the issue may be closed without a comment.
-->

**Operating environment/Installation (Hass.io/Docker/pip/etc.):**
<!--
Please provide details about your environment below this line. -->
Hass.io ESPHome addon version 1.14.3

**ESP (ESP32/ESP8266, Board/Sonoff):**
<!--
Please provide details about which ESP you're using below.
-->
ESP8266

**Affected component:**
<!--
Please add the link to the documentation at https://esphome.io/index.html of the component in question.
-->
https://next.esphome.io/components/climate/fujitsu_general.html

**Description of problem:**
Validate reports ok but compilation fails.

**Problem-relevant YAML-configuration entries:**
```yaml
remote_transmitter:
  pin: D2
  carrier_duty_percent: 50%

climate:
  - platform: fujitsu_general
    name: "device3_hvac"

```

**Logs (if applicable):**
<!--
Please copy the debug log here. If possible, also connect to the ESP over USB and copy those logs into the backticks.
-->
```
src/main.cpp: In function 'void setup()':
src/main.cpp:150:86: error: cannot allocate an object of abstract type 'esphome::fujitsu_general::FujitsuGeneralClimate'
   fujitsu_general_fujitsugeneralclimate = new fujitsu_general::FujitsuGeneralClimate();
                                                                                      ^
In file included from src/esphome.h:23:0,
                 from src/main.cpp:3:
src/esphome/components/fujitsu_general/fujitsu_general.h:10:7: note:   because the following virtual functions are pure within 'esphome::fujitsu_general::FujitsuGeneralClimate':
 class FujitsuGeneralClimate : public climate_ir::ClimateIR {
       ^
In file included from src/esphome/components/climate_ir/climate_ir.h:4:0,
                 from src/esphome.h:21,
                 from src/main.cpp:3:
src/esphome/components/remote_base/remote_base.h:210:16: note: 	virtual bool esphome::remote_base::RemoteReceiverListener::on_receive(esphome::remote_base::RemoteReceiveData)
   virtual bool on_receive(RemoteReceiveData data) = 0;
                ^
Compiling /data/device3/.pioenvs/device3/lib4d9/ESP8266WiFi/ESP8266WiFiGeneric.cpp.o
Compiling /data/device3/.pioenvs/device3/lib4d9/ESP8266WiFi/ESP8266WiFiMulti.cpp.o
*** [/data/device3/.pioenvs/device3/src/main.cpp.o] Error 1
========================= [FAILED] Took 70.60 seconds =========================
```

**Additional information and things you've tried:**




<!-- LEAVE THIS LINE AS-IS AND DON'T DELETE IT, OTHERWISE THE ISSUE WILL BE CLOSED AUTOMATICALLY. -->

--- a/esphome/components/climate_ir/climate_ir.h
+++ b/esphome/components/climate_ir/climate_ir.h
@@ -51,6 +51,9 @@ class ClimateIR : public climate::Climate, public Component, public remote_base:
   /// Transmit via IR the state of this climate controller.
   virtual void transmit_state() = 0;
 
+  // Dummy implement on_receive so implementation is optional for inheritors
+  bool on_receive(remote_base::RemoteReceiveData data) override { return false; };
+
   bool supports_cool_{true};
   bool supports_heat_{true};
   bool supports_dry_{false};

--- a/tests/test1.yaml
+++ b/tests/test1.yaml
@@ -667,7 +667,7 @@ sensor:
       name: "Windspeed"
     wind_direction_degrees:
       name: "Winddirection Degrees"
-    pin: 
+    pin:
       number: GPIO04
       mode: INPUT
   - platform: zyaura
@@ -1082,7 +1082,7 @@ light:
           name: "Test For Custom Lambda Effect"
           lambda: |-
             it[0] = current_color;
-                
+
     - automation:
         name: Custom Effect
         sequence:
@@ -1154,6 +1154,11 @@ climate:
     sensor: my_sensor
   - platform: coolix
     name: Coolix Climate
+  - platform: fujitsu_general
+    name: Fujitsu General Climate
+  - platform: yashima
+    name: Yashima Climate
+
 
 switch:
   - platform: gpio
@@ -1351,7 +1356,7 @@ interval:
             static uint16_t btn_left_state = id(btn_left)->get_value();
 
             ESP_LOGD("adaptive touch", "___  Touch Pad '%s' (T%u): val: %u state: %u tres:%u", id(btn_left)->get_name().c_str(), id(btn_left)->get_touch_pad(), id(btn_left)->get_value(), btn_left_state, id(btn_left)->get_threshold());
-            
+
             btn_left_state = ((uint32_t) id(btn_left)->get_value()  +  63 * (uint32_t)btn_left_state) >> 6;
 
             id(btn_left)->set_threshold(btn_left_state * 0.9);

